### PR TITLE
[CARBONDATA-2324] Support config ExecutorService in search mode

### PR DESCRIPTION
**Pull request by kevinjmh**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4893/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4362/

**Comment by jackylk** [comment]
retest this please

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 1]
```diff
@@ -1635,12 +1635,15 @@
...
 
   /**
-   * Num of threads used in query executor when using search mode.
+   * Thread size of static ExecutorService in each Node when using search mode.
+   * Default value is -1, it means that Executors.newCachedThreadPool() will be used to
+   * maximize utilization. If thread numbers has to be limited, set it a positive Integer
+   * will call Executors.newFixedThreadPool(int nThreads) instead
    */
   @CarbonProperty
   public static final String CARBON_SEARCH_MODE_THREAD = "carbon.search.mode.thread";
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: please rename it to `CARBON_SEARCH_MODE_SCAN_THREAD`, it means the number of thread used to do the scan. There are other threads in thread mode, like handling the request from master

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4966/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4973/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4979/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4425/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**